homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,250 @@
+            CONF_TYPE: ATTR_TYPE_CLOUD,
+        }
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return EzvizOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        # Check if ezviz cloud account is present in entry config.
+        # Return camera user flow if present.
+        for item in self._async_current_entries():
+            if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
+                return await self.async_step_user_camera()
+
+        errors = {}
+
+        if user_input is not None:
+            if CONF_TIMEOUT not in user_input:
+                user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
+
+            try:
+                return await self._validate_and_create_auth(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except AbortFlow:
+                raise
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors or {}
+        )
+
+    async def async_step_user_camera(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[ATTR_SERIAL])
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title=user_input[ATTR_SERIAL],
+                data={
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_TYPE: ATTR_TYPE_CAMERA,
+                },
+            )
+
+        camera_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=DEFAULT_CAMERA_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Required(ATTR_SERIAL): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user_camera", data_schema=camera_schema, errors=errors or {}
+        )
+
+    async def async_step_discovery(self, discovery_info):
+        """Handle a flow for discovered camera without rtsp config entry."""
+
+        await self.async_set_unique_id(discovery_info[ATTR_SERIAL])
+        self._abort_if_unique_id_configured()
+
```

Comment:
Maybe set the title context here to better distinguish the flow in the GUI?https://github.com/home-assistant/core/blob/314a5518f1b0043c74581970dceac20547b8615b/homeassistant/components/deconz/config_flow.py#L209

Reply:
I agree. Added.